fb/usage.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2019 Hudson River Trading LLC
+// All rights reserved.
+
+package fb
+
+import "github.com/prometheus/common/log"
+
+type UsageResponse struct {
+	Groups []UsageGroup
+	Users  []UsageUser
+}
+
+type PaginationData struct {
+	ContinuationToken string `json:"continuation_token"`
+	Total             int    `json:"total_item_count"`
+}
+
+type UsageGroup struct {
+	Items          []UsageItemGroup `json:"items"`
+	PaginationInfo PaginationData   `json:"pagination_info"`
+}
+
+type UsageUser struct {
+	Items          []UsageItemUser `json:"items"`
+	PaginationInfo PaginationData  `json:"pagination_info"`
+}
+
+type UsageItemGroup struct {
+	FileSystem             map[string]string `json:"file_system"`
+	FileSystemDefaultQuota int               `json:"file_system_default_quota"`
+	Group                  NameID            `json:"group"`
+	Name                   string            `json:"name"`
+	Quota                  int               `json:"quota"`
+	Usage                  int               `json:"usage"`
+}
+
+type UsageItemUser struct {
+	FileSystem             map[string]string `json:"file_system"`
+	FileSystemDefaultQuota int               `json:"file_system_default_quota"`
+	User                   NameID            `json:"user"`
+	Name                   string            `json:"name"`
+	Quota                  int               `json:"quota"`
+	Usage                  int               `json:"usage"`
+}
+
+type NameID struct {
+	Id   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+func (fbClient FlashbladeClient) Usage() (UsageResponse, error) {
+	endpoint := "1.8/file-systems"
+	var filesystemsResponse FilesystemsResponse
+	err := fbClient.GetJSON(endpoint, nil, &filesystemsResponse)
+
+	if err != nil {
+		log.Errorf("Error while getting JSON on endpoint %s", endpoint, err)
```

Comment:
I tested on an old version of FlashBlade which doesn't have the 1.8 API endpoint, and you get a 404 back and an empty response which isn't logged.

Not a blocker for merging, just an observation - I can look into adding a `log.Error` when the response status code is 404.